File: commands.py
cmd = {}

# [0] Name
# [1] AliasList
# [2] Description
# [3] Function Call



# -- Help --

def help(sender, args):
	print("Hello")


cmd["help"] = [
	"Help", # Name
	("info", "commands", "cmds"), # Alias list
	"This command displays the command list", # Description
	help # Function Call
]

# -- Vote Kick --


def voteKick(sender, args):
	pass


cmd["votekick"] = [
	"VoteKick", # Name
	("kick", "vk"), # Alias List
	"This command creates a vote to kick a user", # Description
	voteKick # Function Call
]

# -- Vote Mute --

def voteMute(sender, args):
	pass


cmd["votemute"] = [
	"VoteMute", # Name
	("mute", "vm"), # Alias list
	"This command creates a vote to mute a user", # Description
	voteMute # Function Call
]

# -- Vote --

File: commands.js
const {MessageEmbed} = require('discord.js');

var cmds = {};

cmds.help = {
	Name : "Help",
	Alias : ["h", "info", "cmds", "commands"],
	Description : "Displays all available commands",
	Run : function(msg, args){
		console.log("Check DMs");

		const embed = new MessageEmbed()
			.setTitle("All server commands")
			.setColor(0x9aedea)
			for(i = 0; i < cmds.length; i++){
				embed.addFields(cmds[i].Name, cmds[i].Description)
			}
		msg.channel.send(embed);
			
	}
}

cmds.votekick = {
	Name : "VoteKick",
	Alias : ["kick", "vk", "votek"],
	Description : "Vote to kick a user",
	Run : function(msg, args){
		console.log("Initiating kick vote");
	}
}

cmds.votemute = {
	Name : "VoteMute",
	Alias : ["mute", "vm", "votem"],
	Description : "Vote to mute a user in Voice Channels",
	Run : function(msg, args){
		console.log("Initiating mute vote");
	}
}

cmds.createtextchannel = {
	Name : "CreateTextChannel",
	Alias : ["ctc", "createtc", "ctextchannel", "ctextc"],
	Description : "Create a new text channel",
	Run : function(msg, args){
		console.log("Initiating creating text channel vote");
	}
}

cmds.givecustomrank = {
	Name : "GiveCustomRank",
	Alias : ["gcr", "givecr", "gcustomrank", "gcustomr", "gcrank"],
	Description : "Creates a new rank for you",
	Run : function(msg, args){
		console.log("Initiating custom rank creation for " + msg.author);
	}
}

module.exports = cmds
